lib/client-assets.php

```diff
@@ -371,5 +371,12 @@ function gutenberg_scripts_and_styles( $hook ) {
 		array( 'wp-components', 'wp-blocks' ),
 		filemtime( gutenberg_dir_path() . 'editor/build/style.css' )
 	);
+
+	global $hook_suffix;
+	/**
+	 * Hook to allow extending of gutenberg editor in plugin/themes.
+	 */
+	do_action( 'block_enqueue_script', $hook_suffix );
```

Comment:
Also, I think this should actually be `block_edit_enqueue_scripts` to make it clear that the scripts and styles being enqueued are for the _editor_ and not for blocks that are being rendered on the frontend.